app/models/concerns/truncatable.rb

```diff
@@ -0,0 +1,30 @@
+module Truncatable
+  extend ActiveSupport::Concern
+
+  private
+
+  def truncate_html(html, max_length, empty_tag_length = 500)
+    doc = Nokogiri::HTML::DocumentFragment.parse(html)
+    accumulated_length = 0
+    truncated_node = nil
+
+    doc.traverse do |node|
+      if accumulated_length >= max_length
+        node.remove unless truncated_node.ancestors.include?(node)
+        next
+      end
+
+      next unless node.children.empty?
+
+      content_length = node.text? ? node.text.length : empty_tag_length
+      if accumulated_length + content_length >= max_length
+        node.content = node.text.truncate(max_length - accumulated_length) if node.text?
```

Comment:
Do we really need to truncate node contents rather than keeping/discarding it as a whole?

Reply:
If we discard it as a whole, diary entry of someone who created only one huge paragraph, won't be shown at all.
If we keep it fully, diary entry of someone who created only one huge paragraph, will overload diary entries list.
That's why I think truncation is the best method in this case.

Reply:
If you write one huge paragraph, yes it's going to be discarded. Don't write one huge paragraph.

The limit of 1000 characters could be increased though.

Reply:
What do you think is a good number for the limit?

Reply:
Truncation was removed and character limit was increased to 2000. Thanks for reviewing.